tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -296,6 +344,43 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 						heapUsedStats.marginOfErrorPercent > args.maxRelativeMarginOfError
 					);
 
+					if (baselineMemoryUsage >= 0 && allowedDeviationBytes >= 0) {
+						// Compare the average heap used to the baseline memory usage
+						const avgHeapUsed = heapUsedStats.arithmeticMean;
+						const lowerBound = baselineMemoryUsage - args.allowedDeviationBytes;
+						const upperBound = baselineMemoryUsage + args.allowedDeviationBytes;
+						// Throw errors on regressions/improvements if `ENABLE_MEM_REGRESSION` is set and a warning otherwise.
+						// This allows us to run the tests in CI without failing the build, but still get a warning.
+						if (avgHeapUsed > upperBound) {
+							const message = `Memory Regression detected for test '${
+								testObject.title
+							}': Used '${avgHeapUsed.toPrecision(6)}' bytes, with baseline'${
+								args.baselineMemoryUsage
+							}' and tolerance of '${allowedDeviationBytes} bytes.\n`;
+							if (ENABLE_MEM_REGRESSION) {
+								process.stderr.write(message);
+								// eslint-disable-next-line unicorn/no-process-exit
+								process.exit(1);
+							} else {
+								process.stdout.write(message);
```

Comment:
Try this and confirm it does what I expect? I know we use `chalk` in other places in the tool already, it would be nice if these warnings jumped more in the output. Same for the other warning below.
```suggestion
								process.stdout.write(chalk.yellow(message));
```

"!")

Reply:
Yep, chalk works! Warnings show up in yellow